testing/unit_testing/sort_cohorts_test/test_SortCohorts.pf

```diff
@@ -0,0 +1,425 @@
+      new_node%shorter => head%taller%taller%taller%taller%taller
+      
+      allocate(new_node)
+      new_node%height = heights(8)
+      head%taller%taller%taller%taller%taller%taller%taller => new_node
+      new_node%shorter => head%taller%taller%taller%taller%taller%taller
+    
+      patch%tallest => new_node
+      
+      ! sort cohorts
+      call patch%SortCohorts()
+      
+      ! test that the order is correct
+      i = 1
+      new_node => patch%shortest
+      do while (associated(new_node))
+        @assertEqual(new_node%height, heights(i), tolerance=tol)
+        new_node => new_node%taller
+        i = i + 1
+      end do
+           
+    end subroutine SmallList_AscendingOrder_Unchanged
+    
+    @Test 
+    subroutine SmallList_DescendingOrder_Reversed(this)
+      ! test that for a hard-coded list in descending order is sorted, the order remains unchanged
+      class(TestSortCohorts),  intent(inout) :: this ! patch test object
+      type(fates_patch_type),  pointer :: patch
+      type(fates_cohort_type), pointer :: head, new_node
+      real(r8)                         :: heights(8) = (/25.0_r8, 20.0_r8, 12.5001_r8,   &
+        12.5_r8, 12.0_r8, 10.0_r8, 5.0_r8, 2.0_r8/)
+      integer                          :: i
+      
+      ! create a hardcoded doubly linked list
+      allocate(patch)
+      
+      allocate(head)
+      patch%shortest => head
+      head%height = heights(1)
+      
+      allocate(new_node)
+      new_node%height = heights(2)
+      head%taller => new_node
+      new_node%shorter => head
+      
+      allocate(new_node)
+      new_node%height = heights(3)
+      head%taller%taller => new_node
+      new_node%shorter => head%taller
+      
+      allocate(new_node)
+      new_node%height = heights(4)
+      head%taller%taller%taller => new_node
+      new_node%shorter => head%taller%taller
+      
+      allocate(new_node)
+      new_node%height = heights(5)
+      head%taller%taller%taller%taller => new_node
+      new_node%shorter => head%taller%taller%taller
+      
+      allocate(new_node)
+      new_node%height = heights(6)
+      head%taller%taller%taller%taller%taller => new_node
+      new_node%shorter => head%taller%taller%taller%taller
+      
+      allocate(new_node)
+      new_node%height = heights(7)
+      head%taller%taller%taller%taller%taller%taller => new_node
+      new_node%shorter => head%taller%taller%taller%taller%taller
+      
+      allocate(new_node)
+      new_node%height = heights(8)
+      head%taller%taller%taller%taller%taller%taller%taller => new_node
+      new_node%shorter => head%taller%taller%taller%taller%taller%taller
+    
+      patch%tallest => new_node
+      
+      ! sort cohorts
+      call patch%SortCohorts()
+      
+      ! test that the order is correct
+      i = size(heights)
+      new_node => patch%shortest
+      do while (associated(new_node))
+        @assertEqual(new_node%height, heights(i), tolerance=tol)
```

Comment:
Here and elsewhere: I don't think you should need a tolerance, since nothing should adjust the heights from their hard-coded values. (But I guess it doesn't really hurt to have this.)

Reply:
I actually assumed the `tolerance` argument was required haha. so that is good to know that it isn't!